### PR TITLE
remove unused _pin_memory property in PinMemory class 

### PR DESCRIPTION
**Pull request by FightingZhen**
Please read through our [contribution guide](https://github.com/pytorch/data/blob/main/CONTRIBUTING.md) prior to
creating your pull request.

- If you are adding a new node, ensure you read that section in the contribution guide, as it includes requirements for
  functionality and testing.

Fixes #{issue number}
- Not related

### Changes
- Add privateuse1 situation support for _pin_memory property in PinMemory class

### CODE REVIEW

**Comment by divyanshk** [comment]
Thanks @FightingZhen for the PR! Can you elaborate more on your use-case for this?

**Comment by andrewkho** [comment]
I assume it's to get parity with this: https://github.com/pytorch/pytorch/blob/main/torch/utils/data/_utils/pin_memory.py#L31 . 

side note @divyanshk we need to do a sync for stateful_dataloader, updating our tests and code to match pytorch core's dataloader v1

**Comment by andrewkho** [comment]
@FightingZhen any ideas on how to test this? Looks like this PR in pytorch/pytorch landed without tests: https://github.com/pytorch/pytorch/commit/3460b2b7d36eb80cd1c30f0c8f6a7cc849bf339b

File: torchdata/nodes/pin_memory.py [hunk 1]
```diff
@@ -115,7 +115,8 @@ def __init__(
         super().__init__()
         self.source = source
         self.snapshot_frequency = snapshot_frequency
-        self._pin_memory = torch.cuda.is_available()
+        self._pin_memory = torch.cuda.is_available() or getattr(torch,
+                                                                torch._C._get_privateuse1_backend_name()).is_available()
```
Comment (andrewkho) [inline torchdata/nodes/pin_memory.py:119]: actually, it doesn't look like self._pin_memory is even used anywhere, was this causing an issue for you?

Comment (FightingZhen) [reply torchdata/nodes/pin_memory.py:119]: Thanks for your reply, I'm trying to test torchtune with torchdata on Ascend device, and find this position not friendly to privateuse1 situations, so I commit this PR :)
I also find that this property is not currently used. I am not sure whether it is a pre-built property or will be abandoned soon?

Comment (andrewkho) [reply torchdata/nodes/pin_memory.py:119]: How's your experience with Ascend devices? I haven't tried it before. This property was just copied from torch.utils.data and can likely be dropped

Comment (FightingZhen) [reply torchdata/nodes/pin_memory.py]: > How's your experience with Ascend devices? I haven't tried it before. This property was just copied from torch.utils.data and can likely be dropped

If this property can be dropped, my modification is no longer required, I have changed the content in this pr to remove this property, please help me make sure whether this pr is good to go, thanks :) @andrewkho

**Comment by andrewkho** [comment]
Starting an internal CI run to run GPU tests

File: torchdata/nodes/pin_memory.py (hunk 1 above)
Comment (andrewkho) [reply torchdata/nodes/pin_memory.py]: Thanks @FightingZhen ! If we keep this, we could use it to change this node to just pass-through, though I think I prefer users to handle that themselves so the behaviour is less surprising

**Comment by andrewkho** [comment]
Wonder if there was a change to CI runners that broke determinism, the broken tests seem to be related to controlled randomness. Either way, I ran the tests after deleting and everything seems to pass so we should be good

**Review by andrewkho — APPROVED** [review]
LGTM

**Comment by FightingZhen** [comment]
@andrewkho The CI looks finished, I'm not sure whether the above failed jobs is related with my modification or not, can this PR be merged? :)

**Comment by andrewkho** [comment]
@FightingZhen yup please go ahead and merge, thank you for your contribution!

**Comment by FightingZhen** [comment]
@divyanshk can you help me merge this pr? thanks :)

**Pull request merged**